main/ZmqttDiscovery.ino

```diff
@@ -286,44 +278,52 @@ void createDiscovery(const char* sensor_type,
     sensor["cmd_t"] = command_topic; //command_topic
   }
 
+  StaticJsonDocument<JSON_MSG_BUFFER> jsonDeviceBuffer;
+  JsonObject device = jsonDeviceBuffer.to<JsonObject>();
+  JsonArray identifiers = device.createNestedArray("identifiers");
+
   if (gateway_entity) {
-    StaticJsonDocument<JSON_MSG_BUFFER> jsonDeviceBuffer;
-    JsonObject device = jsonDeviceBuffer.to<JsonObject>();
-    char JSONmessageBuffer[JSON_MSG_BUFFER];
-    serializeJson(modules, JSONmessageBuffer, sizeof(JSONmessageBuffer));
-    device["name"] = gateway_name;
-    device["model"] = JSONmessageBuffer;
+    //device representing the board
+    String model = "";
+    serializeJson(modules, model);
+    device["name"] = String(gateway_name);
```

Comment:
Not sure of the benefit of converting this to a string?

Reply:
no apparent reason, it simply seems safer in case of gateway_name is corrupted.